pkg/generators/openapi.go

```diff
@@ -353,10 +353,49 @@ func (g openAPITypeWriter) generateCall(t *types.Type) error {
 	return g.Error()
 }
 
+func (g openAPITypeWriter) generateValueValidations(vs *ValueValidations) {
+
+	if vs == nil {
+		return
+	}
+	if vs.Minimum != nil {
+		g.Do("Minimum: util.PointerToFloat64($.Minimum$),\n", vs)
+	}
+	if vs.Maximum != nil {
+		g.Do("Maximum: util.PointerToFloat64($.Maximum$),\n", vs)
+	}
+	if vs.MinLength != nil {
+		g.Do("MinLength: util.PointerToInt64($.MinLength$),\n", vs)
+	}
+	if vs.MaxLength != nil {
+		g.Do("MaxLength: util.PointerToInt64($.MaxLength$),\n", vs)
+	}
+	if vs.Pattern != nil {
+		g.Do("Pattern: util.PointerToString($.Pattern$),\n", vs)
+	}
+	if vs.MultipleOf != nil {
+		g.Do("MultipleOf: util.PointerToFloat64($.MultipleOf$),\n", vs)
+	}
+	if vs.MinItems != nil {
+		g.Do("MinItems: util.PointerToInt64($.MinItems$),\n", vs)
+	}
+	if vs.MaxItems != nil {
+		g.Do("MaxItems: util.PointerToInt64($.MaxItems$),\n", vs)
+	}
+	if vs.UniqueItems != nil {
+		g.Do("UniqueItems: true,\n", nil)
+	}
+}
+
 func (g openAPITypeWriter) generate(t *types.Type) error {
 	// Only generate for struct type and ignore the rest
 	switch t.Kind {
 	case types.Struct:
+		overrides, err := ParseCommentTags(t.CommentLines)
```

Comment:
Should we also validate the provided tags? Are there other validations that might be reasonable to have of the marker comments?

- Max{X} > Min{X}
- Pattern requires string field, etc.
- Optional, Required, or Defaulted (future PR)

/cc @apelisse @thockin

Reply:
Yeah, we could either verify these here or somewhere else, but to keep things simple I would just plain fail here.

I'll just say, in general the OpenAPI/json-schema is fairly loose with what one can do. so it's totally acceptable to have:
```
{
   maximum: 5
   maxLength: 10
}
```
which means, this can be of any type, but if it's a number it can't be bigger than 5 and if it's a string it can't be longer than 10. If it's an array, or map, it can be of arbitrary length. That being said, there are definitely things that still don't make sense like ...
```
{
   type: string
   maximum: 5
}
```
And we're a lot more likely to have these use-cases, so I would just fail these.